JonathanCoco/angular/product_w_factories/index.html

```diff
@@ -0,0 +1,103 @@
+<html>
+<head>
+  <title></title>
+  <script src='bower_components/angular/angular.js'></script>
+  <script type='text/javascript'>
+    var myApp = angular.module('myApp', []);
+
+    myApp.factory('productFactory', function (){
+        // The factory is nothing more than a function that returns an object
+        var products = [
+            {product_name: 'basketball hoop', product_price: 34.11},
+            {product_name: 'football jersey', product_price: 99.45},
+            {product_name: 'soccer goal', product_price: 250.00}];
+
+        var factory = {};
+        // Add a getStudents key to the factory object with a value of a function.
+        factory.getProducts = function (callback){
+
+            callback(products);
+        }
+
+        factory.addProduct = function(productName, productPrice, callback){
+
+            product = { product_name: productName, product_price: productPrice};
+
+            products.push(product);
+
+            callback(products);
+
+        }
+
+        factory.deleteProduct = function(index, callback){
+
+            products.splice(index, 1);
+            callback(products);
+
+        }
+        // Most important step: return the object so it can be used by the rest of our angular code
+        return factory;
+    });
+
+
+    myApp.controller('productController', ['$scope', 'productFactory', function ($scope, productFactory){
+      // we don't need a factory for this demo, it wouldn't make a difference if we did.
+        $scope.sortType     = 'product_name'; // set the default sort type
+        $scope.sortReverse  = false;  // set the default sort order
+
+        $scope.products = [];
+
+        productFactory.getProducts(function(data) {
+            $scope.products = data
+        });
+
+        $scope.addProduct = function(productName, productPrice){
+
+            productFactory.addProduct(productName, productPrice, function(data) { $scope.products = data })
```

Comment:
same comment as above, since you have access to data through scope from the view, you could save it into a single object and just pass that one object to the factory.